src/Mvc/Mvc.DataAnnotations/src/DataAnnotationsMetadataProvider.cs

```diff
@@ -443,10 +484,51 @@ internal static bool IsNonNullable(IEnumerable<object> attributes)
                 flags.Length >= 0 &&
                 flags[0] == 1) // First element is the property/parameter type.
             {
-                return true;
+                isNullable = true;
+                return true; // [Nullable] found and type is an NNRT
             }
 
-            return false;
+            isNullable = false;
+            return true; // [Nullable] found but type is not an NNRT
+        }
+
+        internal static bool IsNullableBasedOnContext(Type containingType, MemberInfo member)
+        {
+            var attributes = member?.GetCustomAttributes(inherit: true) ?? Array.Empty<object>();
```

Comment:
According to https://github.com/dotnet/roslyn/blob/master/docs/features/nullable-metadata.md, NullableContextAttribute is not inherited.